Ct rocketchat

hDw1z opened the pull request.

Commit:
ct_mongodb
Changes to be committed:
	new file:   ct_mongodb/LICENSE
	new file:   ct_mongodb/defaults/main.yml
	new file:   ct_mongodb/meta/main.yml
	new file:   ct_mongodb/tasks/main.yml

Commit:
ct_mongodb
Changes to be committed:
	modified:   README.md
	new file:   roles/ct_mongodb/README.md
	modified:   roles/ct_mongodb/defaults/main.yml
	modified:   roles/ct_mongodb/tasks/main.yml

Commit:
ct_mongodb
Changes to be committed:
	modified:   roles/ct_mongodb/tasks/main.yml

Commit:
ct_mongodb
Changes to be committed:
	modified:   roles/ct_mongodb/README.md
	modified:   roles/ct_mongodb/defaults/main.yml
	modified:   roles/ct_mongodb/tasks/main.yml

Commit:
ct_mongodb
Changes to be committed:
	modified:   roles/ct_mongodb/defaults/main.yml
	modified:   roles/ct_mongodb/meta/main.yml
	modified:   roles/ct_mongodb/tasks/main.yml
	new file:   roles/ct_mongodb/templates/mongodb.env.j2

Commit:
ct_rocketchat
Changes to be committed:
	new file:   roles/ct_rocketchat/LICENSE
	new file:   roles/ct_rocketchat/README.md
	new file:   roles/ct_rocketchat/defaults/main.yml
	new file:   roles/ct_rocketchat/meta/main.yml
	new file:   roles/ct_rocketchat/tasks/main.yml
	new file:   roles/ct_rocketchat/templates/rocketchat.env.j2

Commit:
ct_rocketchat
Changes to be committed:
	modified:   roles/ct_rocketchat/defaults/main.yml

Commit:
ct_rocketchat
Changes to be committed:
	modified:   README.md

Commit:
ct_rocketchat
Changes to be committed:
	modified:   roles/ct_rocketchat/defaults/main.yml

Commit:
ct_rocketchat
Changes to be committed:
	modified:   roles/ct_rocketchat/defaults/main.yml
	modified:   roles/ct_rocketchat/tasks/main.yml

Commit:
ct_rocketchat
Changes to be committed:
	modified:   roles/ct_rocketchat/defaults/main.yml

Commit:
ct_rocketchat
Changes to be committed:
	modified:   roles/ct_rocketchat/defaults/main.yml

Commit:
ct_rocketchat
Changes to be committed:
	modified:   roles/ct_rocketchat/README.md

Commit:
ct_rocketchat
 Changes to be committed:
	deleted:    roles/ct_mongodb/LICENSE
	deleted:    roles/ct_mongodb/README.md
	deleted:    roles/ct_mongodb/defaults/main.yml
	deleted:    roles/ct_mongodb/meta/main.yml
	deleted:    roles/ct_mongodb/tasks/main.yml
	deleted:    roles/ct_mongodb/templates/mongodb.env.j2

Commit:
ct_rocketchat
Changes to be committed:
	modified:   roles/ct_rocketchat/tasks/main.yml

Commit:
Merge remote-tracking branch 'origin/master' into ct_rocketchat

ct_rocketchat
Changes to be committed:
	modified:   README.md
	new file:   README_BACKUP_10683.md
	new file:   README_BASE_10683.md
	new file:   README_LOCAL_10683.md
	new file:   README_REMOTE_10683.md
	new file:   roles/ct_mongodb/LICENSE
	new file:   roles/ct_mongodb/README.md
	new file:   roles/ct_mongodb/defaults/main.yml
	new file:   roles/ct_mongodb/meta/main.yml
	new file:   roles/ct_mongodb/tasks/main.yml
	new file:   roles/ct_mongodb/templates/mongodb.env.j2

Commit:
ct_rocketchat
 Changes to be committed:
	deleted:    README_BACKUP_10683.md
	deleted:    README_BASE_10683.md
	deleted:    README_LOCAL_10683.md
	deleted:    README_REMOTE_10683.md

ussrlongbow commented on README.md:
```diff
@@ -31,6 +31,12 @@ Configuration of cron jobs. Please consult with role's [README](./roles/cron/REA
 
 ### almaops.common.ct_mongodb
 Deployment a MongoDB container based on the official docker image (https://hub.docker.com/_/mongo). Please consult with role's [README](./roles/ct_mongodb/README.md)
+<<<<<<< HEAD
```
Убрать, артефакт остался от мержа

ussrlongbow commented on README.md:
```diff
@@ -31,6 +31,12 @@ Configuration of cron jobs. Please consult with role's [README](./roles/cron/REA
+### almaops.common.ct_rocketchat
+Deployment a Rocket.Chat container based on the official docker image (https://hub.docker.com/_/rocket-chat). Please consult with role's [README](./roles/ct_rocketchat/README.md)
+=======
+>>>>>>> origin/master
```
Убрать, артефакт остался от мержа

ussrlongbow commented on README.md:
```diff
@@ -40,10 +46,17 @@ Flushing Ansible handlers between other roles. Please consult with role's [READM
 
 ### almaops.common.pip
 Python pip packages installation. Please consult with role's [README](./roles/pip/README.md)
+<<<<<<< HEAD
```
Убрать, артефакт остался от мержа

ussrlongbow commented on README.md:
```diff
@@ -40,10 +46,17 @@ Flushing Ansible handlers between other roles. Please consult with role's [READM
+### almaops.common.pkg
+Packages installation. Please consult with role's [README](./roles/pkg/README.md)
+
+=======
```
Убрать, артефакт остался от мержа

ussrlongbow commented on README.md:
```diff
@@ -40,10 +46,17 @@ Flushing Ansible handlers between other roles. Please consult with role's [READM
 ### almaops.common.pkg
 Packages installation. Please consult with role's [README](./roles/pkg/README.md)
 
+>>>>>>> origin/master
```
Убрать, артефакт остался от мержа

ussrlongbow commented on roles/ct_rocketchat/README.md:
```diff
@@ -0,0 +1,39 @@
+# Role variables
+Please refer to [defaults/main.yml](./defaults/main.yml) for full list of available variables. 
+
+# Dependencies
```
Какие зависимости и зачем?

ussrlongbow commented on roles/ct_rocketchat/defaults/main.yml:
```diff
@@ -0,0 +1,52 @@
+# Path settings - Data directory
+ct_rocketchat_path_data_base: "/srv/ct_rocketchat"
+ct_rocketchat_path_data_volume: "{{ ct_rocketchat_path_data_base }}/{{ ct_rocketchat_ct_name }}"
+ct_rocketchat_path_data: "/app"
```
Видимо должно было быть `{{ ct_rocketchat_path_data_base }}/app`

hDw1z replied:
Не, всё ок. Делал по аналогии с ролью "ct_mongodb".

ussrlongbow replied:
Нет, это не тоже самое
все пути в переменных `<role_name>_path_*` должны быть абсолютными

ussrlongbow commented on roles/ct_rocketchat/defaults/main.yml:
```diff
@@ -0,0 +1,52 @@
+ct_rocketchat_path_ct_data: "/app"
+
+# Network settings
+# ct_rocketchat_bind_addr: "127.0.0.1"
+# ct_rocketchat_bind_port: "3000"
+# ct_rocketchat_bind_ct_port: "3000"
```
Зачем закомментировано?

ussrlongbow commented on roles/ct_rocketchat/defaults/main.yml:
```diff
@@ -0,0 +1,52 @@
+# Container settings - general
+# ct_rocketchat_ct_version: "latest"
+ct_rocketchat_ct_image: "rocket.chat:{{ ct_rocketchat_ct_version }}"
+ct_rocketchat_ct_name: "rocket.chat-{{ ct_rocketchat_ct_version }}"
```
В названии контейнера не должно быть версии

ussrlongbow commented on roles/ct_rocketchat/defaults/main.yml:
```diff
@@ -0,0 +1,52 @@
+# ct_rocketchat_env_port: "3000"
+# ct_rocketchat_env_root_url: "https://rocketchat.company.ru"
+ct_rocketchat_env_mongo_url: "mongodb://{{ ct_bitnami_mongodb_ct_name }}:{{ ct_bitnami_mongodb_bind_port }}/rocketchat?replicaSet={{ ct_bitnami_mongodb_env_mongodb_replica_set_name }}"
+ct_rocketchat_env_mongo_oplog_url: "mongodb://{{ ct_bitnami_mongodb_ct_name }}:{{ ct_bitnami_mongodb_bind_port }}/local?replicaSet={{ ct_bitnami_mongodb_env_mongodb_replica_set_name }}"
```
Здесь не должно быть зависимостий от переменных других ролей, потому что MongoDB может быть развёрнута иным способом

ussrlongbow commented on roles/ct_rocketchat/tasks/main.yml:
```diff
@@ -0,0 +1,75 @@
+    recurse: yes
+    mode: 0744
+    owner: 999
+    group: 999
```
`owner` и `group` должны определяться через переменные

Commit:
ct_rocketchat
 Changes to be committed:
	modified:   README.md
	modified:   roles/ct_rocketchat/README.md
	modified:   roles/ct_rocketchat/defaults/main.yml
	modified:   roles/ct_rocketchat/tasks/main.yml

hDw1z commented:
Замечания устранены. Добавлен [коммит](https://github.com/almaops/ansible-collection-common/pull/3/commits/fcf1f32a2e4b8784361f3a09112c501e1d3ae54a).

ussrlongbow commented on roles/ct_rocketchat/meta/main.yml:
```diff
@@ -0,0 +1,15 @@
+galaxy_info:
+  author: Aleksandr Vzhesnevskiy
+  role_name: ct_rocketchat
+  description: Corporate chat of the company "{{ ct_rocketchat_company_name }}". The role was created based on the "rocket.chat" docker image from https://hub.docker.com
```
Зачем в описании роли использовать переменную? надо укоротить

например
`Role deploys a Rocketchat container from official Docker Hub image`

ussrlongbow commented on roles/ct_rocketchat/defaults/main.yml:
```diff
@@ -0,0 +1,60 @@
+
+# Environment variables
+# ct_rocketchat_env_port: "3000"
+# ct_rocketchat_env_root_url: "https://rocketchat.company.ru"
```
Вообще принято использовать example.com в качестве домена

ussrlongbow replied:
То есть `https://rocketchat.example.com`

ussrlongbow commented on roles/ct_rocketchat/defaults/main.yml:
```diff
@@ -0,0 +1,60 @@
+ct_rocketchat_ct_name: "rocket.chat"
+ct_rocketchat_ct_volume_data:
+  - "{{ ct_rocketchat_path_data_marketplace }}:{{ ct_rocketchat_path_ct_data_marketplace }}"
+  - "{{ ct_rocketchat_path_data_uploads }}/{{ ct_rocketchat_path_ct_data_uploads }}"
```
Опечатка, должно быть `:` вместо `/`

ussrlongbow commented on roles/ct_rocketchat/defaults/main.yml:
```diff
@@ -0,0 +1,60 @@
+ct_rocketchat_ct_networks: []
+# ct_rocketchat_mongodb_addr: "{{ ct_bitnami_mongodb_ct_name }}"
+# ct_rocketchat_mongodb_bind_port: "{{ ct_bitnami_mongodb_bind_port }}"
+# ct_rocketchat_mongodb_replica_set_name: "{{ ct_bitnami_mongodb_env_mongodb_replica_set_name }}"
```
Убрать ссылки на переменные из другой роли, в значении переменной должны быть примеры значений

Commit:
ct_rocketchat
 Changes to be committed:
	modified:   roles/ct_rocketchat/defaults/main.yml
	modified:   roles/ct_rocketchat/meta/main.yml
	modified:   roles/ct_rocketchat/templates/rocketchat.env.j2

hDw1z commented:
Замечания устранены. Добавлено 2 коммита. Последний [коммит](https://github.com/almaops/ansible-collection-common/pull/3/commits/c1d7368f99c32a0477b2909502284d4f159eea17).

Commit:
ct_rocketchat
 Changes to be committed:
	modified:   roles/ct_rocketchat/defaults/main.yml

ussrlongbow commented on roles/ct_rocketchat/templates/rocketchat.env.j2:
```diff
@@ -0,0 +1,22 @@
+MONGO_OPLOG_URL={{ ct_rocketchat_env_mongo_oplog_url }}
+DEPLOY_METHOD={{ ct_rocketchat_env_deploy_method }}
+Accounts_UseDNSDomainCheck={{ ct_rocketchat_env_accounts_usednsdomaincheck }}
+OVERWRITE_SETTING_LDAP_Enable={{ ct_rocketchat_env_overwrite_setting_ldap_enable }}
+OVERWRITE_SETTING_LDAP_Host={{ ct_rocketchat_env_overwrite_setting_ldap_host }}
+OVERWRITE_SETTING_LDAP_Port={{ ct_rocketchat_env_overwrite_setting_ldap_port }}
+OVERWRITE_SETTING_LDAP_Reconnect={{ ct_rocketchat_env_overwrite_setting_ldap_reconnect }}
+OVERWRITE_SETTING_LDAP_Authentication={{ ct_rocketchat_env_overwrite_setting_ldap_authentication }}
+OVERWRITE_SETTING_LDAP_Authentication_UserDN={{ ct_rocketchat_env_overwrite_setting_ldap_authentication_userdn }}
+OVERWRITE_SETTING_LDAP_Authentication_Password={{ ct_rocketchat_env_overwrite_setting_ldap_authentication_password }}
+OVERWRITE_SETTING_LDAP_BaseDN={{ ct_rocketchat_env_overwrite_setting_ldap_basedn }}
+OVERWRITE_SETTING_LDAP_User_Search_Filter={{ ct_rocketchat_env_overwrite_setting_ldap_user_search_filter }}
+OVERWRITE_SETTING_LDAP_User_Search_Scope={{ ct_rocketchat_env_overwrite_setting_ldap_user_search_scope }}
+OVERWRITE_SETTING_LDAP_User_Search_Field={{ ct_rocketchat_env_overwrite_setting_ldap_user_search_field }}
+OVERWRITE_SETTING_LDAP_Default_Domain={{ ct_rocketchat_env_overwrite_setting_ldap_default_domain }}
+OVERWRITE_SETTING_LDAP_Username_Field={{ ct_rocketchat_env_overwrite_setting_ldap_username_field }}
```
вот весь этот блок имеет смысл обособить оператором if

примерно так:
```
{% if ct_rocketchat_env_overwrite_setting_ldap_enable %}
OVERWRITE_SETTING_LDAP_***
...
{% endif %}
```

ussrlongbow commented on roles/ct_rocketchat/defaults/main.yml:
```diff
@@ -0,0 +1,72 @@
+ct_rocketchat_env_mongo_oplog_url: "mongodb://{{ ct_rocketchat_mongodb_addr }}:{{ ct_rocketchat_mongodb_bind_port }}/local?replicaSet={{ ct_rocketchat_mongodb_replica_set_name }}"
+ct_rocketchat_env_deploy_method: "docker"
+ct_rocketchat_env_accounts_usednsdomaincheck: "false"
+# ct_rocketchat_env_overwrite_setting_ldap_enable: "true"
```
здесь поставить дефолтное значение false и убрать из комментария

Commit:
ct_rocketchat
 Changes to be committed:
	modified:   roles/ct_rocketchat/defaults/main.yml
	modified:   roles/ct_rocketchat/templates/rocketchat.env.j2

hDw1z commented:
Замечания устранены. Добавлен [коммит](https://github.com/almaops/ansible-collection-common/pull/3/commits/0410503d35cdc134aaa49e8ea38b8acb84385cb3).

ussrlongbow reviewed (approved):
вроде всё ок
